src/App.tsx

```diff
@@ -1,15 +1,52 @@
-import React from 'react';
+import React, { useMemo, useState, useCallback } from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
+import classNames from 'classnames';
+import debounce from 'lodash.debounce';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [filterValue, setFilterValue] = useState('');
+  const [debouncedFilterValue, setDebouncedFilterValue] = useState(filterValue);
+
+  const debouncedSetFilterValue = useCallback(
+    debounce((value: string) => {
+      setDebouncedFilterValue(value);
+    }, 300),
+    [],
+  );
+
+  const handleSelectedPerson = (person: Person | null) => {
+    setSelectedPerson(person);
+    setFilterValue(person ? person.name : '');
+  };
+
+  const handleChangeText = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const value = event.target.value;
+
+    setFilterValue(value);
+
+    debouncedSetFilterValue(value);
+
+    if (selectedPerson && value) {
+      setSelectedPerson(null);
+    }
+  };
+
+  const filteredPeople = useMemo(() => {
+    return peopleFromServer.filter(user =>
+      user.name.toLowerCase().includes(debouncedFilterValue.toLowerCase()),
+    );
+  }, [debouncedFilterValue]);
```

Comment:
Ensure that the filter won't be called if the user entered spaces only. Currently, if `debouncedFilterValue` is just spaces, the filter function will still run, which might not be the intended behavior. Consider trimming the value and checking if it's not empty before proceeding with the filter operation.